src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,168 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ELEMENTDATA = {};
+    private static final Object[] DEFAULTCAPACITY_EMPTY_ELEMENTDATA = {};
+    private Object[] elementData;
+    private int size = 0;
+
+    public ArrayList() {
+        this.elementData = new Object[]{};
+
+    }
+
+    public ArrayList(int customCapacity) {
+        if (customCapacity > 0) {
+            this.elementData = new Object[customCapacity];
+        } else if (customCapacity == 0) {
+            this.elementData = EMPTY_ELEMENTDATA;
+        } else {
+            throw new RuntimeException("Impossible capacity: "
+                    + customCapacity);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
```

Comment:
The growth logic should be extracted into a separate method to adhere to the Single Responsibility Principle. The `add` method is responsible for adding an element and should not directly handle the array resizing.